solidity/contracts/middleware/InterchainAccountRouter.sol

```diff
@@ -650,15 +813,20 @@ contract InterchainAccountRouter is Router {
      * @param _owner The remote owner of the interchain account
      * @param _router The remote origin InterchainAccountRouter
      * @param _ism The local address of the ISM
+     * @param _userSalt Salt provided by the user, allows control over account derivation.
      * @return The CREATE2 salt used for deploying the interchain account
      */
     function _getSalt(
         uint32 _origin,
         bytes32 _owner,
         bytes32 _router,
-        bytes32 _ism
+        bytes32 _ism,
+        bytes32 _userSalt
     ) private pure returns (bytes32) {
-        return keccak256(abi.encodePacked(_origin, _owner, _router, _ism));
+        return
+            keccak256(
+                abi.encodePacked(_origin, _owner, _router, _ism, _userSalt)
```

Comment:
I think this is the right tradeoff but just noting this means already deployed proxies are incompatible with this derivation and we should not roll this out as an ICA upgrade